cmake/OpenCVInstallLayout.cmake

```diff
@@ -0,0 +1,122 @@
+# Following macro updates multiple variables at once:
+#  - OPENCV_BIN_INSTALL_PATH
+#  - OPENCV_LIB_INSTALL_PATH
+#  - OPENCV_CONFIG_INSTALL_PATH
+#  - ...
+macro(ocv_set_install_layout)
+  unset(_var)
+  foreach(arg ${ARGN})
+    if(NOT _var)
+      set(_var ${arg})
+    else()
+      ocv_update(OPENCV_${_var}_INSTALL_PATH ${arg})
+      unset(_var)
+    endif()
+  endforeach()
+endmacro()
+
+if(OPENCV_CUSTOM_INSTALL_LAYOUT)
```

Comment:
?
```.diff
-OPENCV_CUSTOM_INSTALL_LAYOUT
+OPENCV_CUSTOM_INSTALL_LAYOUT_FILE
```

It would be nice to add message of applying of that file (to track possible typos somehow - we can't rely on CMake due `ocv_cmake_dump_vars()` dump).